### PR TITLE
Add guava validation not null

### PR DESCRIPTION
**Pull request by AndreyWolox**
## Summary
- Add guava precondiction to the setters of the models
- Add guava dependence
## Trello Card
https://trello.com/c/4j3ojn2V/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

### CODE REVIEW

File: src/main/java/wolox/training/models/User.java [hunk 1]
```diff
@@ -42,37 +45,40 @@ public User() {
...
+        this.name = checkNotNull(name, ValidationError.NULL_VALUE.getMsg());
     }
 
     public LocalDate getBirthdate() {
         return birthdate;
     }
 
     public void setBirthdate(LocalDate birthdate) {
-        this.birthdate = birthdate;
+        this.birthdate = checkNotNull(birthdate, ValidationError.NULL_VALUE.getMsg());
```
Comment (leandrocostawolox) [inline src/main/java/wolox/training/models/User.java]: You should check if the birthdate is before now :)

File: src/main/java/wolox/training/models/Book.java [hunk 2]
```diff
@@ -56,78 +59,78 @@ public long getId() {
...
+        this.year = checkNotNull(year, ValidationError.NULL_VALUE.getMsg());
     }
 
     public int getPages() {
         return pages;
     }
 
     public void setPages(int pages) {
-        this.pages = pages;
+        this.pages = checkNotNull(pages, ValidationError.NULL_VALUE.getMsg());
```
Comment (leandrocostawolox) [inline src/main/java/wolox/training/models/Book.java]: The pages shouldn't be null, or 0 either

File: src/main/java/wolox/training/models/Book.java [hunk 3]
```diff
@@ -56,78 +59,78 @@ public long getId() {
...
+        this.title = checkNotNull(title, ValidationError.NULL_VALUE.getMsg());
     }
 
     public String getSubtitle() {
         return subtitle;
     }
 
     public void setSubtitle(String subtitle) {
-        this.subtitle = subtitle;
+        this.subtitle = checkNotNull(subtitle, ValidationError.NULL_VALUE.getMsg());
```
Comment (leandrocostawolox) [inline src/main/java/wolox/training/models/Book.java]: For Strings it's good to check for null or empty

**Review by leandrocostawolox — CHANGES_REQUESTED** [review]
Some recommendations about field validations :)

**Review by leandrocostawolox — APPROVED** [review]

File: src/main/java/wolox/training/models/Book.java [hunk 4]
```diff
@@ -56,14 +61,15 @@ public long getId() {
     }
 
     public void setId(long id) {
-        this.id = id;
+        this.id = checkNotNull(id, ValidationError.NULL_VALUE.getMsg());
```
Comment (CarlosLugo22) [inline src/main/java/wolox/training/models/Book.java:64]: Great, a comment this is one of a lot preconditions, for example for id just be numbers, or if is a email needs to have and special character like '@', or for names not have special characters if they are not needed and for the years or birhdate that the date is in a range or something like that.

**Review by CarlosLugo22 — APPROVED** [review]
Good, a comment just to have in mind

**Pull request merged**